packages/coordinator/src/services/translation/request/dispense/dispense-notification.ts

```diff
@@ -329,16 +329,18 @@ function createSuppliedLineItem(
   const fhirPrescriptionDispenseItemNumber = getPrescriptionItemNumber(fhirMedicationDispense)
   const globalIdentifier = new hl7V3.GlobalIdentifier(fhirPrescriptionDispenseItemNumber)
   const pertinentInformation2 = createSuppliedLineItemPertinentInformation2(fhirMedicationDispense)
-  return new hl7V3.DispenseNotificationSuppliedLineItem(globalIdentifier, pertinentInformation2)
+  return pertinentInformation2
+    ? new hl7V3.DispenseNotificationSuppliedLineItemNotDispened(globalIdentifier, pertinentInformation2)
+    : new hl7V3.DispenseNotificationSuppliedLineItem(globalIdentifier)
```

Comment:
Instead of having another class for the case of non-dispensed medication, could we use the same approach we have in `dispense-claim.ts` within coordinator/.../translation/request/dispense, e.g.
```ts
  const nonDispensingReason = getSuppliedLineItemNonDispensingReason(detail)
  if (nonDispensingReason) {
    suppliedLineItem.pertinentInformation2 = new hl7V3.SuppliedLineItemPertinentInformation2(nonDispensingReason)
  }
```

Reply:
The code has been refactored
```ts
function createSuppliedLineItem(
  fhirMedicationDispense: fhir.MedicationDispense
): hl7V3.DispenseNotificationSuppliedLineItem {
  const fhirPrescriptionDispenseItemNumber = getPrescriptionItemNumber(fhirMedicationDispense)
  const suppliedLineItem = new hl7V3.DispenseNotificationSuppliedLineItem(
    new hl7V3.GlobalIdentifier(fhirPrescriptionDispenseItemNumber)
  )

  const nonDispensingReason = getFhirStatusReasonCodeableConceptCode(fhirMedicationDispense)
  if (nonDispensingReason) {
    const pertInformation2 = new hl7V3.NonDispensingReason(
      nonDispensingReason.code,
      nonDispensingReason.display
    )
    suppliedLineItem.pertinentInformation2 = new hl7V3.PertinentInformation2NonDispensing(pertInformation2)
  }

  return suppliedLineItem
}
```